codec/reflectcodec/struct_fielder.go

```diff
@@ -18,13 +18,18 @@ const (
 
 	// TagValue is the value the tag must have to be serialized.
 	TagValue = "true"
+
+	// TagValue is the value the tag must have to be serialized, this variant
+	// includes Omit Empty
+	TagWithOmitEmptyValue = "true,omitempty"
```

Comment:
"omitempty" has the effect of omitting a value in other serializations (eg, json) when the value is empty/nil, so I would expect eg, adding "omitempty" to a field's serialization spec to reduce the size of the serialized output.

I'd prefer a name like "allownils" or "preservenil" or "withnil" or "nullable" that wouldn't have the above connotation (unless it's not possible to use a different name here)